composer.json

```diff
@@ -42,7 +42,7 @@
         "drupal/persistent_login": "^1.0-alpha4",
         "drupal/phingdrushtask": "dev-7.x-2.x#f4b020f09da3c174fce40f2bd9a9617689f57124",
         "drupal/piwik": "~1.0",
-        "drupal/piwik_reporting_api": "^1.0-beta2",
+        "drupal/piwik_reporting_api": "dev-1.x",
```

Comment:
In the upstream, I would  try to refactor PiwikQueryFactory:
As it is a factory, all methods should be static to communicate that the factory itself is stateless.

Reply:
Do you think we should open an upstream ticket there?